src/base_maths.jl

```diff
@@ -83,39 +83,45 @@ function _check_matmul_dims(A::AbstractMatrix, B::AbstractMatrix)
     ))
 end
 
-function Base.:*(B::BlockDiagonal, M::AbstractMatrix)
+function Base.:*(B::BlockDiagonal{T}, M::AbstractMatrix) where T
```

Comment:
Does this do what you expect for empty matrices? We should test for that if we aren't already. Sometimes broadcast does a bad thing in that case.

Reply:
What do you mean by "empty matrices"?

Reply:
like `Matrix(undef, 0, 0)`?

Reply:
More like `zeros(0, 0)` or zeros(5, 0)` but yeah

Reply:
added tests for these cases :)